src/Utilities/Functional.hpp

```diff
@@ -0,0 +1,255 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines function objects
```

Comment:
Just remove the file comment. It's not required and I don't think it's useful enough here.